src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java

```diff
@@ -248,92 +253,157 @@ private void generateC8DocumentsForRespondents(Map<String, Object> updatedCaseDa
                 callbackRequest,
                 respondent
             );
-            String partyName = respondent.getValue().getLabelForDynamicList();
-            Document c8FinalDocument = null;
-            Document c8FinalWelshDocument = null;
-            if (dataMap.containsKey(IS_CONFIDENTIAL_DATA_PRESENT)) {
-                dataMap.put("dynamic_fileName", partyName
-                    + " " + LocalDateTime.now().format(dateTimeFormatter) + ".pdf");
-                c8FinalDocument = documentGenService.generateSingleDocument(
-                    authorisation,
-                    caseData,
-                    C8_RESP_FINAL_HINT,
-                    false,
-                    dataMap
-                );
-                dataMap.put("dynamic_fileName", partyName
-                    + " " + LocalDateTime.now().format(dateTimeFormatter) + "welsh" + ".pdf");
-                c8FinalWelshDocument = documentGenService.generateSingleDocument(
-                    authorisation,
-                    caseData,
-                    C8_RESP_FINAL_HINT,
-                    true,
-                    dataMap
-                );
-            }
-            populateC8Documents(updatedCaseData, caseData, c8FinalDocument, c8FinalWelshDocument, respondentIndex);
+            populateC8Documents(authorisation,
+                        updatedCaseData,
+                        caseData,
+                        dataMap, checkIfDetailsChanged(callbackRequest,respondent),
+                        respondentIndex,respondent
+            );
             respondentIndex++;
         }
     }
 
-    private  void populateC8Documents(Map<String, Object> updatedCaseData, CaseData caseData,
-                                      Document c8FinalDocument,
-                                      Document c8WelshDocument, int partyIndex) {
-        if (null != c8FinalDocument && partyIndex >= 0) {
-            ResponseDocuments c8ResponseDocuments = ResponseDocuments.builder()
-                .dateTimeCreated(LocalDateTime.now())
-                .build();
+    public Boolean checkIfDetailsChanged(CallbackRequest callbackRequest, Element<PartyDetails> respondent) {
```

Comment:
change the method name please.

Reply:
changed to checkIfConfidentialityDetailsChangedRespondent